### PR TITLE
Update protobuf and gRPC related dependencies

### PR DESCRIPTION
**Pull request by rolinh**
See commits for details.

Note: I wanted to bump `grpc-go` to v1.31.0 but I can't due to a breaking change introduced in [v1.30.0 ](https://github.com/grpc/grpc-go/releases/tag/v1.30.0) ("Remove the deprecated `naming` package.") that in turn breaks `go.etcd.io/etcd` (see https://github.com/etcd-io/etcd/issues/12124 for details).

There's a PR opened for etcd to upgrade `grpc-go` (https://github.com/etcd-io/etcd/pull/12155) but it's stalled due to other breaking changes introduced in a minor version of `grpc-go` which breaks the API.

### CODE REVIEW

**Comment by Rolinh** [comment]
test-me-please

**Review by aanm — APPROVED** [review]

File: go.mod [hunk 1]
```diff
@@ -50,7 +50,7 @@ require (
 	github.com/kr/pretty v0.2.0
 	github.com/mattn/go-shellwords v1.0.5
 	github.com/miekg/dns v1.0.14
-	github.com/mitchellh/protoc-gen-go-json v0.0.0-20200113165135-fd297ce346f1
+	github.com/mitchellh/protoc-gen-go-json v0.0.0-20200717004536-8fbb6f3d7291
```
Comment (sayboras) [inline go.mod:53]: just a note, this tool is still using deprecated `github.com/golang/protobuf` package as well.

Comment (rolinh) [reply go.mod:53]: It's not the whole of `github.com/golang/protobuf` that is deprecated but just `github.com/golang/protobuf/protoc-gen-go`.

And apparently only `github.com/envoyproxy/protoc-gen-validate` still imports it as stated in the commit message:
```
$ go mod why "github.com/golang/protobuf/protoc-gen-go"
# github.com/golang/protobuf/protoc-gen-go
github.com/cilium/cilium/pkg/hubble/tools
github.com/envoyproxy/protoc-gen-validate
github.com/golang/protobuf/protoc-gen-go
```

Comment (sayboras) [reply go.mod:53]: You are right, thanks :100: 

Actually, I meant this deprecated package `github.com/golang/protobuf/jsonpb` which is used in protoc-gen-go-json.

Comment (rolinh) [reply go.mod:53]: Yes, there's an issue for this but it's still open: https://github.com/mitchellh/protoc-gen-go-json/issues/6

**Comment by Rolinh** [comment]
All CI tests passed but somehow the GA tests didn't trigger. I'll mark this PR as draft and back to see if that works (I can't trigger them manually).

**Comment by tklauser** [comment]
test-me-please

**Review by tklauser — APPROVED** [review]

**Review by kaworu — APPROVED** [review]

**Comment by tklauser** [comment]
retest-net-next

**Comment by tklauser** [comment]
retest-runtime

**Comment by tklauser** [comment]
retest-gke

**Comment by Rolinh** [comment]
retest-runtime

**Pull request merged**